src/NumericalAlgorithms/Spectral/ComplexDataView.cpp

```diff
@@ -5,56 +5,83 @@
 
 #include "DataStructures/ComplexDataVector.hpp"
 #include "DataStructures/DataVector.hpp"
+#include "ErrorHandling/Assert.hpp"
 #include "Utilities/Gsl.hpp"
 #include "Utilities/Math.hpp"  // IWYU pragma: keep
 
 namespace Spectral {
 namespace SWSH {
 namespace detail {
 
+template <typename T>
+void sizes_assert(const T& vector, const size_t size) {
+  ASSERT(vector.size() == size,
+         "Assignment must be to the same size,"
+         " not "
+             << vector.size() << " assigned to " << size);
+}
+
+template void sizes_assert<DataVector>(const DataVector&, const size_t);
+template void sizes_assert<ComplexDataVector>(const ComplexDataVector&,
+                                              const size_t);
+template void sizes_assert<ComplexDataView<ComplexRepresentation::Interleaved>>(
+    const ComplexDataView<ComplexRepresentation::Interleaved>&, const size_t);
+template void
+sizes_assert<ComplexDataView<ComplexRepresentation::RealsThenImags>>(
+    const ComplexDataView<ComplexRepresentation::RealsThenImags>&,
+    const size_t);
+
 // Interleaved constructors do not cause copies or allocations
 
-// clang-tidy: This class uses manual memory management deliberately,
-// so silence complaints about pointer math and casts
+// Due to the nature of this class performing manual memory management, indexing
+// into complex data with perscribed strides, it is necessary to violate type
+// system conventions to index into the complex data as though it is double
+// data, thus using a `reinterpret_cast`. The C++ standard guarantees that
+// complex numbers are represented in memory in such a way that this is correct,
+// despite the type subversion. We also cannot take the address of the
+// components of the complex data, as those are returned by rvalue.
```

Comment:
`rvalue` -> `value, not lvalue reference` but overall, this is much better, thank you :)